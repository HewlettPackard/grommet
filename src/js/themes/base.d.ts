import { DeepReadonly, ColorType, OpacityType, BackgroundType } from '../utils'

export declare const base: DeepReadonly<ThemeType>;
export declare const generate: (baseSpacing: number, scale: number) => DeepReadonly<ThemeType>;

declare const colors: {
  active: ColorType;
  black: ColorType;
  border: ColorType;
  brand: ColorType;
  control: ColorType;
  focus: ColorType;
  placeholder: ColorType;
  selected: ColorType;
  text: ColorType;
  icon: ColorType;
  white: ColorType;
};
  
type Colors = typeof colors & {
  'accent-1': string;
  'accent-2': string;
  'accent-3': string;
  'accent-4': string;
  'neutral-1': string;
  'neutral-2': string;
  'neutral-3': string;
  'neutral-4': string;
  'dark-1': string;
  'dark-2': string;
  'dark-3': string;
  'dark-4': string;
  'dark-5': string;
  'dark-6': string;
  'light-1': string;
  'light-2': string;
  'light-3': string;
  'light-4': string;
  'light-5': string;
  'light-6': string;
  'status-critical': string;
  'status-error': string;
  'status-warning': string;
  'status-ok': string;
  'status-unknown': string;
  'status-disabled': string;
};

export interface ThemeType {
  global: {
    animation: {
      duration: string;
      jiggle: {
        duration: string;
      };
    };
    borderSize: {
      xsmall: string;
      small: string;
      medium: string;
      large: string;
      xlarge: string;
    };
    breakpoints: {
      small: {
        value: number;
        borderSize: {
          xsmall: string;
          small: string;
          medium: string;
          large: string;
          xlarge: string;
        };
        edgeSize: {
          none: string;
          hair: string;
          xxsmall: string;
          xsmall: string;
          small: string;
          medium: string;
          large: string;
          xlarge: string;
        };
        size: {
          xxsmall: string;
          xsmall: string;
          small: string;
          medium: string;
          large: string;
          xlarge: string;
          full: string;
        };
      };
      medium: {
        value: number;
      };
      large: {};
    };
    deviceBreakpoints: {
      phone: string;
      tablet: string;
      computer: string;
    };
    colors: Colors;
    control: {
      border: {
        width: string;
        radius: string;
        color: ColorType;
      };
    };
    debounceDelay: number;
    drop: {
      background: BackgroundType;
      border: {
        width: string;
        radius: string;
      };
      shadowSize: string;
      zIndex: string;
    };
    edgeSize: {
      none: string;
      hair: string;
      xxsmall: string;
      xsmall: string;
      small: string;
      medium: string;
      large: string;
      xlarge: string;
      responsiveBreakpoint: string;
    };
    elevation: {
      light: {
        none: string;
        xsmall: string;
        small: string;
        medium: string;
        large: string;
        xlarge: string;
      };
      dark: {
        none: string;
        xsmall: string;
        small: string;
        medium: string;
        large: string;
        xlarge: string;
      };
    };
    focus: {
      border: {
        color: ColorType;
      };
    };
    font: {
      size: string;
      height: string;
      maxWidth: string;
    };
    hover: {
      background: {
        dark: {
          color: ColorType;
          opacity: OpacityType;
        };
        light: {
          color: ColorType;
          opacity: OpacityType;
        };
      };
      color: ColorType;
    };
    input: {
      padding: string;
      weight: number;
    };
    opacity: {
      strong: number;
      medium: number;
      weak: number;
    };
    selected: {
      background: string;
      color: ColorType;
    };
    spacing: string;
    size: {
      xxsmall: string;
      xsmall: string;
      small: string;
      medium: string;
      large: string;
      xlarge: string;
      xxlarge: string;
      full: string;
    };
  };
  accordion: {
    icons: {
      collapse: any;
      expand: any;
    };
  };
  anchor: {
    textDecoration: string;
    fontWeight: number;
    color: ColorType;
    hover: {
      textDecoration: string;
    };
  };
  box: {
    responsiveBreakpoint: string;
  };
  button: {
    border: {
      width: string;
      radius: string;
    };
    primary: {};
    disabled: {
      opacity: number;
    };
    minWidth: string;
    maxWidth: string;
    padding: {
      vertical: string;
      horizontal: string;
    };
  };
  calendar: {
    small: {
      fontSize: string;
      lineHeight: number;
      daySize: string;
      slideDuration: string;
    };
    medium: {
      fontSize: string;
      lineHeight: number;
      daySize: string;
      slideDuration: string;
    };
    large: {
      fontSize: string;
      lineHeight: number;
      daySize: string;
      slideDuration: string;
    };
    icons: {
      previous: any;
      next: any;
      small: {
        previous: any;
        next: any;
      };
    };
  };
  carousel: {
    icons: {
      current: any;
      next: any;
      previous: any;
    };
  };
  checkBox: {
    border: {
      color: ColorType;
      width: string;
    };
    check: {
      extend: string;
      radius: string;
      thickness: string;
    };
    extend: string;
    color: ColorType;
    icon: { size: string; extend: string };
    icons: {};
    hover: {
      border: {
        color: ColorType;
      };
    };
    size: string;
    toggle: {
      color: ColorType;
      background: {
        dark: string;
        light: string;
      };
      radius: string;
      size: string;
      knob: { extend: string };
    };
  };
  clock: {
    analog: {
      hour: {
        color: ColorType;
        width: string;
        size: string;
        shape: string;
      };
      minute: {
        color: ColorType;
        width: string;
        size: string;
        shape: string;
      };
      second: {
        color: ColorType;
        width: string;
        size: string;
        shape: string;
      };
      size: {
        small: string;
        medium: string;
        large: string;
        xlarge: string;
        huge: string;
      };
    };
    digital: {
      text: {
        xsmall: {
          size: string;
          height: number;
        };
        small: {
          size: string;
          height: number;
        };
        medium: {
          size: string;
          height: number;
        };
        large: {
          size: string;
          height: number;
        };
        xlarge: {
          size: string;
          height: number;
        };
        xxlarge: {
          size: string;
          height: number;
        };
      };
    };
  };
  collapsible: {
    minSpeed: number;
    baseline: number;
  };
  dataTable: {
    header: {};
    groupHeader: {
      border: {
        side: string;
        size: string;
      };
      fill: string;
      pad: {
        horizontal: string;
        vertical: string;
      };
      background: {
        dark: string;
        light: string;
      };
    };
    icons: {
      ascending: any;
      contract: any;
      descending: any;
      expand: any;
    };
    resize: {
      border: {
        side: string;
        color: ColorType;
      };
    };
    primary: {
      weight: string;
    };
  };
  formField: {
    border: {
      color: ColorType;
      position: string;
      side: string;
      error: {
        color: ColorType;
      };
    };
    error: {
      color: ColorType;
    };
    help: {
      color: ColorType;
    };
    label: {};
  };
  grommet: {};
  heading: {
    font: {};
    level: {
      1: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      2: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      3: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      4: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      5: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
      6: {
        font: {};
        small: {
          size: string;
          height: string;
          maxWidth: string;
        };
        medium: {
          size: string;
          height: string;
          maxWidth: string;
        };
        large: {
          size: string;
          height: string;
          maxWidth: string;
        };
        xlarge: {
          size: string;
          height: string;
          maxWidth: string;
        };
      };
    };
    responsiveBreakpoint: string;
    weight: number;
  };
  icon: {
    size: {
      small: string;
      medium: string;
      large: string;
      xlarge: string;
    };
  };
  layer: {
    background: string;
    border: {
      radius: string;
    };
    container: {
      zIndex: string;
    };
    overlay: {
      background: string;
    };
    responsiveBreakpoint: string;
    zIndex: string;
  };
  menu: {
    icons: {
      down: any;
    };
  };
  paragraph: {
    small: {
      size: string;
      height: string;
      maxWidth: string;
    };
    medium: {
      size: string;
      height: string;
      maxWidth: string;
    };
    large: {
      size: string;
      height: string;
      maxWidth: string;
    };
    xlarge: {
      size: string;
      height: string;
      maxWidth: string;
    };
    xxlarge: {
      size: string;
      height: string;
      maxWidth: string;
    };
  };
  radioButton: {
    border: {
      color: ColorType;
      width: string;
    };
    check: {
      radius: string;
    };
    hover: {
      border: {
        color: ColorType;
      };
    };
    icon: {};
    icons: {};
    gap: string;
    size: string;
  };
  rangeInput: {
    track: {
      height: string;
      color: any;
    };
    thumb: {};
  };
  select: {
    container: {};
    control: {};
    icons: {
      down: any;
    };
    step: number;
  };
  tab: {
    active: {
      color: ColorType;
    };
    border: {
      side: string;
      size: string;
      color: ColorType;
      active: {
        color: ColorType;
      };
      hover: {
        color: ColorType;
      };
    };
    color: ColorType;
    hover: {
      color: ColorType;
    };
    margin: {
      vertical: string;
      horizontal: string;
    };
    pad: {
      bottom: string;
    };
  };
  tabs: {
    header: {};
    panel: {};
  };
  table: {
    header: {
      align: string;
      pad: {
        horizontal: string;
        vertical: string;
      };
      border: string;
      verticalAlign: string;
      fill: string;
    };
    body: {
      align: string;
      pad: {
        horizontal: string;
        vertical: string;
      };
    };
    footer: {
      align: string;
      pad: {
        horizontal: string;
        vertical: string;
      };
      border: string;
      verticalAlign: string;
      fill: string;
    };
  };
  text: {
    xsmall: {
      size: string;
      height: string;
      maxWidth: string;
    };
    small: {
      size: string;
      height: string;
      maxWidth: string;
    };
    medium: {
      size: string;
      height: string;
      maxWidth: string;
    };
    large: {
      size: string;
      height: string;
      maxWidth: string;
    };
    xlarge: {
      size: string;
      height: string;
      maxWidth: string;
    };
    xxlarge: {
      size: string;
      height: string;
      maxWidth: string;
    };
  };
  video: {
    captions: {
      background: string;
    };
    icons: {
      closedCaption: any;
      configure: any;
      fullScreen: any;
      pause: any;
      play: any;
      reduceVolume: any;
      volume: any;
    };
  };
  worldMap: {
    color: ColorType;
    continent: {
      active: string;
      base: string;
    };
    hover: {
      color: ColorType;
    };
    place: {
      active: string;
      base: string;
    };
  };
}
