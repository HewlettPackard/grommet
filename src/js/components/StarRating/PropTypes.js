import PropTypes from 'prop-types';

let PropType = {};
if (process.env.NODE_ENV !== 'production') {
  PropType = {
    color: PropTypes.string,
    scale: PropTypes.number,
  };
}
export const FeedbackRatingTypes = PropType;
