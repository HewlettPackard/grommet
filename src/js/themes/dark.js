import { rgba } from 'polished';
import { css } from 'styled-components';

import { deepFreeze } from '../utils/object';
import { normalizeColor } from '../utils/colors';

const controlColor = '#FFCA58';
const accentColors = ['#FD6FFF', '#60EB9F', '#60EBE1', '#FFCA58'];
const neutralColors = ['#EB6060', '#01C781', '#6095EB', '#FFB200'];
const statusColors = {
  critical: '#FF3333',
  error: '#FF3333',
  warning: '#F7E464',
  ok: '#7DD892',
  unknown: '#a8a8a8',
  disabled: '#a8a8a8',
};
const backgroundColor = '#111111';
const textColor = '#eeeeee';
const borderColor = rgba(255, 255, 255, 0.33);
const activeColor = rgba(102, 102, 102, 0.5);

const colors = {
  active: activeColor,
  background: backgroundColor,
  black: '#000000',
  border: borderColor,
  brand: '#FD6FFF',
  control: controlColor,
  focus: controlColor,
  placeholder: '#AAAAAA',
  text: textColor,
  white: '#FFFFFF',
};

const colorArray = (array, prefix) =>
  array.forEach((color, index) => {
    colors[`${prefix}-${index + 1}`] = color;
  });

colorArray(accentColors, 'accent');
colorArray(neutralColors, 'neutral');
Object.keys(statusColors).forEach(color => {
  colors[`status-${color}`] = statusColors[color];
});

export const dark = deepFreeze({
  global: {
    colors,
    drop: {
      background: '#333333',
    },
    focus: {
      border: {
        color: css`
          ${props => normalizeColor('focus', props.theme)};
        `,
        width: '2px',
      },
    },
    font: {
      family: 'Arial',
    },
    input: {
      weight: 700,
    },
    text: {
      dark: textColor,
      light: '#000000',
    },
  },
  anchor: {
    color: controlColor,
  },
  icon: {
    color: textColor,
    colors,
  },
  layer: {
    background: backgroundColor,
    overlay: {
      background: rgba(48, 48, 48, 0.5),
    },
  },
});
