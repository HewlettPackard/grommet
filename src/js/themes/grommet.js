import { css } from 'styled-components';

import { deepFreeze } from '../utils';

const workSansPath = 'https://fonts.gstatic.com/s/worksans/v2';
const zillaSlabPath = 'https://fonts.gstatic.com/s/zillaslab/v3';

export const grommet = deepFreeze({
  global: {
    font: {
      family: "'Work Sans', Arial, sans-serif",
      face: `
        @font-face {
          font-family: 'Work Sans';
          font-style: normal;
          font-weight: 400;
          src:
            local('Work Sans'),
            local('WorkSans-Regular'),
            url("${workSansPath}/ElUAY9q6T0Ayx4zWzW63VJBw1xU1rKptJj_0jans920.woff2") format('woff2');
        }

        @font-face {
          font-family: 'Work Sans';
          font-style: normal;
          font-weight: 600;
          src:
            local('Work Sans SemiBold'),
            local('WorkSans-SemiBold'),
            url("${workSansPath}/z9rX03Xuz9ZNHTMg1_ghGRampu5_7CjHW5spxoeN3Vs.woff2") format('woff2');
        }

        @font-face {
          font-family: 'Zilla Slab';
          font-style: normal;
          font-weight: 400;
          src:
            local('Zilla Slab Regular'),
            local('ZillaSlab-Regular'),
            url("${zillaSlabPath}/dFa6ZfeM_74wlPZtksIFajo6_V6LVlA.woff2") format('woff2');
        }

        @font-face {
          font-family: 'Zilla Slab';
          font-style: normal;
          font-weight: 600;
          src:
            local('Zilla Slab SemiBold'),
            local('ZillaSlab-SemiBold'),
            url("${zillaSlabPath}/dFa5ZfeM_74wlPZtksIFYuUe6HOpW3pwfa0.woff2") format('woff2');
        }
      `,
    },
  },
  anchor: {
    textDecoration: 'underline',
    color: {
      dark: '#9060EB',
      light: '#9060EB',
    },
  },
  button: {
    extend: css`
      ${props => !props.plain && 'font-weight: bold;'}
    `,
  },
  checkBox: {
    icon: {
      extend: css`
        box-sizing: border-box;
        position: absolute;
        top: 0px;
        left: 0px;
        width: ${props => props.theme.checkBox.size};
        height: ${props => props.theme.checkBox.size};
      `,
    },
  },
  heading: {
    font: {
      family: "'Zilla Slab', 'Work Sans', Arial, sans-serif",
    },
  },
});
