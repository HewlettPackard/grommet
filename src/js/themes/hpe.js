import { deepFreeze } from '../utils';

const accentColors = ['#2AD2C9', '#614767', '#ff8d6d'];
const neutralColors = ['#425563', '#5F7A76', '#80746E', '#767676'];
const statusColors = {
  critical: '#F04953',
  error: '#F04953',
  warning: '#FFD144',
  ok: '#01a982',
  unknown: '#CCCCCC',
  disabled: '#CCCCCC',
};

const colors = {
  accent: accentColors,
  brand: '#01a982',
  focus: accentColors[0],
  neutral: neutralColors,
  status: statusColors,
};

const colorArray = (array, prefix) =>
  array.forEach((color, index) => {
    colors[`${prefix}-${index + 1}`] = color;
  });

colorArray(accentColors, 'accent');
colorArray(neutralColors, 'neutral');
Object.keys(statusColors).forEach((color) => {
  colors[`status-${color}`] = statusColors[color];
});

export default deepFreeze({
  global: {
    colors,
    font: {
      family: "'Metric', Arial, sans-serif",
      face: `
        @font-face {
          font-family: "Metric";
          src: url("https://hpefonts.s3.amazonaws.com/web/MetricHPE-Web-Regular.woff") format('woff');
        }

        @font-face {
          font-family: "Metric";
          src: url("https://hpefonts.s3.amazonaws.com/web/MetricHPE-Web-Bold.woff") format('woff');
          font-weight: 700;
        }

        @font-face {
          font-family: "Metric";
          src: url("https://hpefonts.s3.amazonaws.com/web/MetricHPE-Web-Semibold.woff") format('woff');
          font-weight: 600;
        }

        @font-face {
          font-family: "Metric";
          src: url("https://hpefonts.s3.amazonaws.com/web/MetricHPE-Web-Light.woff") format('woff');
          font-weight: 100;
        }
      `,
    },
  },
  anchor: {
    color: 'inherit',
  },
  button: {
    border: {
      radius: '0px',
    },
    colors: {
      accent: '#ff8d6d',
      secondary: 'rgba(51,51,51,0.6)',
    },
    extend: 'letter-spacing: 0.04167em;',
  },
  clock: {
    second: {
      color: {
        dark: '#01a982',
        light: '#01a982',
      },
    },
  },
  heading: {
    font: false,
  },
});
