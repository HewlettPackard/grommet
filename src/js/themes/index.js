import base from './base';

export { default as base } from './base';
export { default as grommet } from './grommet';
export { default as hpe } from './hpe';
export { default as dark } from './dark';

export default base;
