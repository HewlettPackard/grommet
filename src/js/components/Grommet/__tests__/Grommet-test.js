import React, { Component } from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { hpe as hpeTheme } from '../../../themes';

import { Grommet } from '../';

Enzyme.configure({ adapter: new Adapter() });

class TestAnnouncer extends Component {
  static contextTypes = {
    grommet: PropTypes.object,
  }
  componentDidMount() {
    this.context.grommet.announce('hello', 'assertive');
  }
  render() {
    return <div>hi</div>;
  }
}

test('Grommet renders', () => {
  const component = renderer.create(
    <Grommet />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Grommet hpe theme renders', () => {
  const component = renderer.create(
    <Grommet theme={hpeTheme}>Grommet App</Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Grommet full renders', () => {
  const component = renderer.create(
    <Grommet full={true}>Grommet App</Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Grommet announces', (done) => {
  const component = mount(
    <Grommet><TestAnnouncer /></Grommet>
  );
  expect(component.getDOMNode()).toMatchSnapshot();

  // no style, no need for expectPortal
  expect(document.body.querySelector('[aria-live]')).toMatchSnapshot();

  setTimeout(() => {
    // should clear the aria-live container
    expect(document.body.querySelector('[aria-live]')).toMatchSnapshot();
    done();
  }, 600); // wait the aria-live container to clear
});
