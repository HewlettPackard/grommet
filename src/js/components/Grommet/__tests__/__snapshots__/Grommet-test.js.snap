// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grommet announces 1`] = `
<div
  class="StyledGrommet-gKbdxL fqqdJc"
>
  <div>
    hi
  </div>
</div>
`;

exports[`Grommet announces 2`] = `
<div
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
  aria-live="assertive"
>
  hello
</div>
`;

exports[`Grommet announces 3`] = `
<div
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
  aria-live="assertive"
/>
`;

exports[`Grommet hpe theme renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  Grommet App
</div>
`;

exports[`Grommet renders 1`] = `
.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
/>
`;
